### PR TITLE
[feat] get step of sequence api

### PR DESCRIPTION
**Pull request by devfox-se**

### CODE REVIEW

**Comment by devfox-se** [comment]
Implements #1566

File: aim/web/api/runs/object_views.py [hunk 1]
```diff
@@ -95,6 +95,38 @@ async def sequence_batch_api(run_id: str,
...
+            api = CustomObjectApi(seq_name, resolve_blobs=cls.resolve_blobs)
+            api.set_dump_data_fn(cls.dump_record_fn)
+            api.set_requested_traces(run, requested_traces)
+
+            if record_step is None:
+                total_record_range = api.get_total_record_range()
+                if api.use_list:
+                    total_record_range = total_record_range[0]
+                record_step = total_record_range.stop - 1
+            record_range = checked_range(f"{record_step}:{record_step + 1}")
```
Comment (alberttorosyan) [inline aim/web/api/runs/object_views.py:124]: why convert to string then back to tuple?
`record_range = IndexRange(record_step, record_step + 1)`

File: aim/web/api/runs/object_views.py [hunk 2]
```diff
@@ -95,6 +95,38 @@ async def sequence_batch_api(run_id: str,
...
+            run = repo.get_run(run_id)
+            if not run:
+                raise HTTPException(status_code=404)
+
+            api = CustomObjectApi(seq_name, resolve_blobs=cls.resolve_blobs)
+            api.set_dump_data_fn(cls.dump_record_fn)
+            api.set_requested_traces(run, requested_traces)
+
+            if record_step is None:
+                total_record_range = api.get_total_record_range()
```
Comment (alberttorosyan) [inline aim/web/api/runs/object_views.py:120]: total range will be calculated twice. 1st time here, 2nd time after `set_ranges` call. better to leave this logic in `CustomObjectApi` and do the calculation the only once

**Review by alberttorosyan — CHANGES_REQUESTED** [review]
left some comments

File: aim/web/api/runs/object_api_utils.py [hunk 3]
```diff
@@ -86,6 +90,11 @@ def set_ranges(self, record_range: IndexRange, record_density: int,
         else:
             self.total_record_range = self._calculate_ranges()
 
+        if record_step is not None:
+            if record_step < 0:
+                record_step += self.total_record_range.stop
```
Comment (alberttorosyan) [inline aim/web/api/runs/object_api_utils.py:95]: do we have a real use-case for this? if so, still need to check that modulo of record_step is in the allowed range.

Comment (alberttorosyan) [reply aim/web/api/runs/object_api_utils.py:95]: the alternative to specifying -1 as a default would be `None` and checking that ether `record_range` or `record_step` is not None

Comment (devfox-se) [reply aim/web/api/runs/object_api_utils.py:95]: yeah that was the initial method, yet i think its confusing

Comment (devfox-se) [reply aim/web/api/runs/object_api_utils.py:95]: I added assertion just in case

**Review by alberttorosyan — APPROVED** [review]

**Pull request merged**